models/comment.rb

```diff
@@ -91,20 +92,34 @@ def to_hash(params={})
       self.class.hash_tree(subtree_hash).first
     else
       as_document.slice(*%w[body course_id endorsed endorsement anonymous anonymous_to_peers created_at updated_at at_position_list])
-          .merge('id' => _id)
-          .merge('user_id' => author_id)
-          .merge('username' => author_username)
-          .merge('depth' => depth)
-          .merge('closed' => comment_thread.nil? ? false : comment_thread.closed)
-          .merge('thread_id' => comment_thread_id)
-          .merge('parent_id' => parent_ids[-1])
-          .merge('commentable_id' => comment_thread.nil? ? nil : comment_thread.commentable_id)
-          .merge('votes' => votes.slice(*%w[count up_count down_count point]))
-          .merge('abuse_flaggers' => abuse_flaggers)
-          .merge('type' => 'comment')
+          .merge("id" => _id)
+          .merge("user_id" => author_id)
+          .merge("username" => author_username)
+          .merge("depth" => depth)
+          .merge("closed" => comment_thread.nil? ? false : comment_thread.closed) # ditto
+          .merge("thread_id" => comment_thread_id)
+          .merge("parent_id" => parent_ids[-1])
+          .merge("commentable_id" => comment_thread.nil? ? nil : comment_thread.commentable_id) # ditto
+          .merge("votes" => votes.slice(*%w[count up_count down_count point]))
+          .merge("abuse_flaggers" => abuse_flaggers)
+          .merge("type" => "comment")
+          .merge("child_count" => get_cached_child_count)
     end
   end
 
+  def get_cached_child_count
+    if self.child_count.nil?
+      self.child_count = update_cached_child_count
```

Comment:
This makes all the code read much better.  Thanks for the update.

nit: I think you can drop `self.child_count = ` because the update method does this.